package StorageLog;

message LogEntry{

    enum Operation{
        WRITE = 0;
        READ = 1;
        ERASE = 2;
    }
    enum MetadataType{
        NO_METADATA = 0;
        YAFFSHEADER = 1;
        YAFFSSPARE = 2;
    }
    required Operation op = 1;
    required uint64 logIndex = 2;
    required uint64 address = 3;
    optional bytes  data = 4;
    optional uint32 length = 7;
    optional MetadataType mType = 5;
    optional bytes metadata = 6;


}
